### ISSUE TITLE
Version Validation

### ISSUE THREAD

**Issue by BenjamenMeyer** [vegastrike/Assets-Masters#19]
How do we make sure the Release Version is the correctly specified?

**Comment by BenjamenMeyer** [comment]
Some discussed options:
- strict release names and CI/CD integration
- validation step in the release process
- unit test validation to see that things match up

We should at least do the validation step during the release processes while we figure this out; that will satisfy requirements for 0.7.x release if we don't have something by then.

NOTE: This is presently marked against the next release (0.7.x) just to keep it at the top of our priorities. It can slip to 0.8.x or 0.9.x if need be.

**Comment by stephengtuggy** [comment]
We have come a long ways with regard to versioning in the last few months. With the 0.8.x release series, we should be able to ensure that the desired version -- including git commit SHA -- is automatically included, correctly, in the installer filenames, installer versions, and code for both the Game Engine and the Assets.

As a next step, I'm thinking maybe we do away with hard-coded version numbers altogether, and just have the relevant parts of the game (like the Introduction and Credits screens) pull the version in code, of the engine and of the game assets, and dynamically display those. How does that sound to you guys?

**Comment by BenjamenMeyer** [comment]
Agreed. We're close to being able to have a single point of version definition. The engine should be really close; most of the work probably needs to be done on the Asset side, but now that the Engine exposes a version to the assets that can be done - we just have to require that the Engine API Version is 1 or greater to do so.

**Comment by BenjamenMeyer** [comment]
@stephengtuggy should we close this? Or should we make sure that the Assets Masters has a copy of the code for checking the Engine Version first?

**Comment by stephengtuggy** [comment]
> [@stephengtuggy](https://github.com/stephengtuggy) should we close this? Or should we make sure that the Assets Masters has a copy of the code for checking the Engine Version first?

@BenjamenMeyer Not sure. I'll have to get back to you on this one.
